testing/src/test/java/io/stargate/it/cql/DataTypeTest.java

```diff
@@ -0,0 +1,265 @@
+    Row row = session.execute(selectQuery).one();
+    assertThat(row).isNotNull();
+    assertThat(row.get(0, sample.javaType)).isEqualTo(sample.value);
+  }
+
+  private static List<TypeSample<?>> generateAllTypes(CqlIdentifier keyspaceId) {
+    List<TypeSample<?>> primitiveTypes =
+        ImmutableList.of(
+            typeSample(DataTypes.ASCII, GenericType.STRING, "sample ascii"),
+            typeSample(DataTypes.BIGINT, GenericType.LONG, Long.MAX_VALUE),
+            typeSample(
+                DataTypes.BLOB, GenericType.of(ByteBuffer.class), Bytes.fromHexString("0xCAFE")),
+            typeSample(DataTypes.BOOLEAN, GenericType.BOOLEAN, Boolean.TRUE),
+            typeSample(DataTypes.DECIMAL, GenericType.BIG_DECIMAL, new BigDecimal("12.3E+7")),
+            typeSample(DataTypes.DOUBLE, GenericType.DOUBLE, Double.MAX_VALUE),
+            typeSample(DataTypes.FLOAT, GenericType.FLOAT, Float.MAX_VALUE),
+            typeSample(DataTypes.INET, GenericType.INET_ADDRESS, sampleInet()),
+            typeSample(DataTypes.TINYINT, GenericType.BYTE, Byte.MAX_VALUE),
+            typeSample(DataTypes.SMALLINT, GenericType.SHORT, Short.MAX_VALUE),
+            typeSample(DataTypes.INT, GenericType.INTEGER, Integer.MAX_VALUE),
+            typeSample(DataTypes.DURATION, GenericType.CQL_DURATION, CqlDuration.from("PT30H20M")),
+            typeSample(DataTypes.TEXT, GenericType.STRING, "sample text"),
+            typeSample(
+                DataTypes.TIMESTAMP, GenericType.INSTANT, Instant.ofEpochMilli(872835240000L)),
+            typeSample(DataTypes.DATE, GenericType.LOCAL_DATE, LocalDate.ofEpochDay(16071)),
+            typeSample(
+                DataTypes.TIME, GenericType.LOCAL_TIME, LocalTime.ofNanoOfDay(54012123450000L)),
+            typeSample(
+                DataTypes.TIMEUUID,
+                GenericType.UUID,
+                UUID.fromString("FE2B4360-28C6-11E2-81C1-0800200C9A66")),
+            typeSample(
+                DataTypes.UUID,
+                GenericType.UUID,
+                UUID.fromString("067e6162-3b6f-4ae2-a171-2470b63dff00")),
+            typeSample(
+                DataTypes.VARINT,
+                GenericType.BIG_INTEGER,
+                new BigInteger(Integer.MAX_VALUE + "000")));
+
+    List<TypeSample<?>> allTypes = new ArrayList<>();
+    for (TypeSample<?> type : primitiveTypes) {
+      // Generate additional samples from each primitive
+      allTypes.add(type);
+      allTypes.add(listOf(type));
+      if (type.javaType != GenericType.CQL_DURATION) {
+        allTypes.add(setOf(type));
+        allTypes.add(mapToIntFrom(type));
+      }
+      allTypes.add(mapOfIntTo(type));
+      allTypes.add(tupleOfIntAnd(type));
+      allTypes.add(udtOfIntAnd(type, keyspaceId));
+    }
+    return allTypes;
+  }
+
+  static class TypeSample<JavaTypeT> {
+
+    static <JavaTypeT> TypeSample<JavaTypeT> typeSample(
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6263918)